packages/runtime/src/types.ts

```diff
@@ -235,14 +237,54 @@ export enum TransactionType {
   CallNoOpSSC,
   ClearSSC,
   CloseSSC,
-  DeleteSSC
+  DeleteSSC,
+  DeployASA,
+  DeploySSC,
+  OptInASA,
+  OptInSSC
 }
 
 export interface Sign {
   sign: SignType
   lsig?: LogicSig
 }
 
+export interface DeployASAParam extends Sign {
+  type: TransactionType.DeployASA
+  fromAccount: AccountSDK
+  asaName: string
+  asaDef?: ASADef
+  payFlags: TxParams
```

Comment:
All transaction parameters have `payFlags`, `fromAccount` common, and also each of them *extends Sign*. 

How about moving these common fields to `Sign` interface? What do you think?

Reply:
i think sign is good, because payFlags seems unrelated

Reply:
I was thinking something like:
```javascript
interface BasicParams extends Sign {
  fromAccount: AccountSDK, // keep common keys here
  payFlags: TxParams
}
```
And all transaction parameters will extend *BasicParams*. Just a thought.

Reply:
good thought